### ISSUE TITLE
Filter on many to many relation where match equals all

### ISSUE THREAD

**Issue by maartenvanbenthem** [directus/directus#2313]
# Feature Request
Adding the possibility to filter collections with a many to many relation

## What problem does this feature solve?
I have a `page` collection and a `tag` collection. I would like to filter pages that have a relation with a `tag` with title values **Tag 1** AND **Tag 2** AND **Tag 3**.

My pages table:
![image](https://user-images.githubusercontent.com/6450927/54278201-0866b000-4592-11e9-99c7-7df8178ac888.png)
My tags table:
![image](https://user-images.githubusercontent.com/6450927/54278228-1a485300-4592-11e9-98fa-2eb61c00ccdc.png)
The many to many relational table (page_tags):
![image](https://user-images.githubusercontent.com/6450927/54278283-3ea42f80-4592-11e9-8f38-ef3dca837d22.png)

## How do you think this should be implemented?
No idea, but the following query is giving me the right result:
```
SELECT p.*
  FROM page p
 WHERE EXISTS (SELECT NULL
                 FROM page_tags tg
                 JOIN tags t ON t.id = tg.tags_id
                WHERE t.title IN ('Tag 1','Tag 2','Tag 3')
                  AND tg.page_id = p.id
             GROUP BY tg.page_id
               HAVING COUNT(DISTINCT t.title) = 3)
```
`HAVING COUNT(DISTINCT t.title) = 3` makes sure I only get a result with ALL values:

![image](https://user-images.githubusercontent.com/6450927/54279314-0fdb8880-4595-11e9-8689-fa756567407c.png)

Here is my DB
[somedb (1).sql.gz](https://github.com/directus/directus/files/2961413/somedb.1.sql.gz)

**Comment by benhaynes** [comment]
Hey @maartenvanbenthem — it seems like this is already an issue. Can you check out this other ticket and see if it's what you're looking for? If it's not I'll re-open this one...

https://github.com/directus/api/issues/576

**Comment by maartenvanbenthem** [comment]
Hi @benhaynes it is not what I am looking for. The `all` and `has` operators can only be applied while filtering on a one to many relationship. And in my case I am using a many to many relationship.

**Comment by benhaynes** [comment]
Yeah, it's an important one. We'll be working on fixing that one next so hopefully it's resolved soon!
